Resources/doc/5-Usage-Event_Listener.md

```diff
@@ -1,75 +1,110 @@
 
-```xml
-<parameters>
-    <parameter key="acme_demo.sitemap.listener.class">Acme\DemoBundle\EventListener\SitemapListener</parameter>
-</parameters>
+Imagine that your application is (or has) a blog, and that you want to add to your sitemap
+all blog posts that your administrator has created.
 
-<services>
-    <service id="my.sitemap.listener" class="%acme_demo.sitemap.listener.class%">
-        <tag name="presta.sitemap.listener" />
-        <argument type="service" id="router"/>
-    </service>
-</services>
-```
-
-or in yaml:
+**note :** we choose an `event subscriber` as example, but you can also do it with an `event listener`.
 
-```yaml
-parameters:
-    acme_demo.sitemap.listener.class: Acme\DemoBundle\EventListener\SitemapListener
 
-services:
-    my.sitemap.listener:
-        class: "%acme_demo.sitemap.listener.class%"
-        arguments: ["@router"]
-        tags: [{name: "presta.sitemap.listener"}]
-```
+## Service configuration
 
-Sitemap listener example `Acme/DemoBundle/EventListener/SitemapListener.php`:
+Implementation example `AppBundle/EventListener/SitemapBlogPostSubscriber.php`:
 
 ```php
 <?php
-namespace Acme\DemoBundle\EventListener;
 
-use Symfony\Component\Routing\RouterInterface;
-use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+namespace AppBundle\EventListener;
 
-use Presta\SitemapBundle\Service\SitemapListenerInterface;
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Routing\RouterInterface;
 use Presta\SitemapBundle\Event\SitemapPopulateEvent;
 use Presta\SitemapBundle\Sitemap\Url\UrlConcrete;
 
-class SitemapListener implements SitemapListenerInterface
+class SitemapBlogPostSubscriber implements EventSubscriberInterface
 {
+    /**
+     * @var RouterInterface
+     */
     private $router;
 
-    public function __construct(RouterInterface $router)
+    /**
+     * @var EntityManager
+     */
+    private $manager;
+
+    /**
+     * @param RouterInterface $router
+     * @param EntityManager   $manager
+     */
+    public function __construct(RouterInterface $router, EntityManager $manager)
     {
         $this->router = $router;
+        $this->manager = $manager;
     }
 
-    public function populateSitemap(SitemapPopulateEvent $event)
+    /**
+     * @inheritdoc
+     */
+    public static function getSubscribedEvents()
     {
-        $section = $event->getSection();
-        if (is_null($section) || $section == 'default') {
-            //get absolute homepage url
-            $url = $this->router->generate('homepage', array(), UrlGeneratorInterface::ABSOLUTE_URL);
+        return [
+            SitemapPopulateEvent::ON_SITEMAP_POPULATE => 'registerBlogPostsPages',
+        ];
+    }
+
+    /**
+     * @param SitemapPopulateEvent $event
+     */
+    public function registerBlogPostsPages(SitemapPopulateEvent $event)
+    {
+        $posts = $this->manager->getRepository('AppBundle:BlogPost')->findAll();
```

Comment:
Can we add note for big results set that better use `iterate` instead of loading all result set http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/batch-processing.html#iterating-results ? Or maybe hydrate as array.